### DISCUSSION TITLE
Arcade Game support Requests

### DISCUSSION THREAD

**Discussion by WilsonBradley** [curif/AgeOfJoy-2022.1#22] [Ideas]
Joust 
Battle Zone
Asteroids

**Comment by WilsonBradley** [comment]
Space Invaders 
Popeye

**Comment by Elleby77** [comment]
Star Wars
Tempest
Zaxxon
Centipede

**Comment by WilsonBradley** [comment]
Joust?

**Comment by WilsonBradley** [comment]
I tried a bunch of Roms in the same folder, but after going into app, did not see arcades for them. I guess I can rename the "test" roms to an official.. Is that the way to do it?

**Reply by curif** [reply]
I don't understand exactly what you are trying to do, but each ROM should have it's own cabinet.
If you create a `test.zip` cabinet to test a ROM, and you want to deploy that cabinet in the gallery, you should rename the cabinet from `test.zip` to other filename and copy it again to the `/cabinets` folder. As a convention I use the same name for both, but that confused some users (logically).


Next citation was extracted from: https://curifab.itch.io/age-of-joy/devlog/456444/about-the-age-of-joy-cabinet-building (and corrected, because I find some errors in the description).

### 4- Deploy the new cabinet

If your game is running without any noticeable issue, then you can deploy the new cabinet:

- Rename the file in your PC from `test.zip` to a file name that represents the game in the cabinet, in our example should be `galaxian.zip` 
- Copy the file to the `/sdcard/Android/data/com.curif.AgeOfJoy/cabinets` folder.
- Start AGE of Joy again, the program will find the new cabinet and move it to the cabinets database (the `/sdcard/Android/data/com.curif.AgeOfJoy/cabinetsdb` folder). Don't touch that folder is for internal uses only. The game will delete the original zip file from `/sdcard/Android/data/com.curif.AgeOfJoy/cabinets`
- After register the new cabinet, it will be deployed in an empty slot inside a Room in the game, replacing an "out of order" cabinet.

**Comment by Elleby77** [comment]
Is there a place where we can upload the cabinets made so we can share with the community?

I just finished a Popeye cab and it works fine (except the screen is a little too low due to the cabinet model (donkeykong) being set for a vertical screen

**Reply by curif** [reply]
Not yet. May be you can set a public folder in drive, then share the link. I will add it to the database.

**Reply by Elleby77** [reply]
Sure, I'll figure something out.

Still have a few more cabs in the DK series I want to make.

**Reply by curif** [reply]
Great! Thank you.

**Comment by WilsonBradley** [comment]
Sorry, not clear on how to test other Roms. I put additional ROMs in the download folder and renamed the cabinets romname.zip and placed this in cabinet. They never showed up to try in game.
joust.zip for example.
Update
Ok, read the above link and will follow, recreating new cabinet from old one

**Comment by WilsonBradley** [comment]
*Moving this post into cabinets/Rom section

Awesome, got the test cabinet to work in the workroom. It's joust.
1.) joust screen needs to be adjusted to landscape from current Portrait layout.
 2.) Note for joust: First time needs to have quarter put in and initialized. After leaving and coming back, it works.
3.) Seriously need ability to assign buttons.

Will work on designing cabinet / assets/ video/sound once it's playable.
